file.hh

```diff
@@ -29,105 +28,101 @@ bool tryPossibleName( std::string const & name, std::string & copyTo );
+  void read( void * buf, qint64 size );
+  void write( void const * buf, qint64 size );
 
-  /// Reads the number of bytes to the buffer, throws an error if it
-  /// failed to fill the whole buffer (short read, i/o error etc).
-  void read( void * buf, qint64 size ) ;
+  // Read sizeof(T) bytes and convert the read data to T
+  template< typename T > T read()
+  {
+    T value;
+    readType( value );
+    return value;
+  }
 
-  template< typename T >
-  void read( T & value ) 
-  { read( &value, sizeof( value ) ); }
-
-  template< typename T >
-  T read() 
-  { T value; read( value ); return value; }
+  template< typename T > void write( T const & value )
+  {
+    write( &value, sizeof( value ) );
+  }
 
   /// Attempts reading at most 'count' records sized 'size'. Returns
   /// the number of records it managed to read, up to 'count'.
-  size_t readRecords( void * buf, qint64 size, size_t count ) ;
-
-  /// Writes the number of bytes from the buffer, throws an error if it
-  /// failed to write the whole buffer (short write, i/o error etc).
-  /// This function employs write buffering, and as such, writes may not
-  /// end up on disk immediately, or a short write may occur later
-  /// than it really did. If you don't want write buffering, use
-  /// writeRecords() function instead.
-  void write( void const * buf, qint64 size ) ;
-
-  template< typename T >
-  void write( T const & value ) 
-  { write( &value, sizeof( value ) ); }
+  size_t readRecords( void * buf, qint64 size, qint64 count );
 
   /// Attempts writing at most 'count' records sized 'size'. Returns
   /// the number of records it managed to write, up to 'count'.
   /// This function does not employ buffering, but flushes the buffer if it
   /// was used before.
-  size_t writeRecords( void const * buf, qint64 size, size_t count )
-    ;
+  size_t writeRecords( void const * buf, qint64 size, qint64 count );
 
-  /// Reads a string from the file. Unlike the normal fgets(), this one
-  /// can strip the trailing newline character, if this was requested.
+  /// Read a line with option to strip the trailing newline character
   /// Returns either s or 0 if no characters were read.
-  char * gets( char * s, int size, bool stripNl = false ) ;
+  char * gets( char * s, int size, bool stripNl = false );
 
-  /// Like the above, but uses its own local internal buffer (1024 bytes
-  /// currently), and strips newlines by default.
-  std::string gets( bool stripNl = true ) ;
+  /// Like the above, but uses its own local internal buffer and strips newlines by default.
+  std::string gets( bool stripNl = true );
 
-  /// Seeks in the file, relative to its beginning.
-  void seek( qint64 offset ) ;
-  uchar * map( qint64 offset, qint64 size );
-  /// Seeks in the file, relative to the current position.
-  void seekCur( qint64 offset ) ;
-  /// Seeks in the file, relative to the end of file.
-  void seekEnd( qint64 offset = 0 ) ;
```

Comment:
I think the seekEnd implementation is intended.

Reply:
Maybe the `offset` is useful in the past, but the `offset` is never used in anywhere in the current codebase.